src/proxy.rs

```diff
@@ -788,34 +793,36 @@ impl Dst for Uri {
 }
 
 static SYS_PROXIES: Lazy<Arc<SystemProxyMap>> =
-    Lazy::new(|| Arc::new(get_sys_proxies(get_from_registry())));
+    Lazy::new(|| Arc::new(get_sys_proxies(get_from_platform())));
 
 /// Get system proxies information.
 ///
-/// It can only support Linux, Unix like, and windows system.  Note that it will always
-/// return a HashMap, even if something runs into error when find registry information in
-/// Windows system.  Note that invalid proxy url in the system setting will be ignored.
+/// All platforms will check for proxy settings via the `http_proxy`, `https_proxy`,
```

Comment:
Environment variables are case sensitive--technically this respects `HTTP_PROXY`, `http_proxy`, `HTTPS_PROXY`, `https_proxy`, `ALL_PROXY`, or `all_proxy`.

Reply:
Rewrote this a bit to not be so exhaustive.